presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp

```diff
@@ -255,11 +256,20 @@ void PrestoExchangeSource::handleDataResponse(
 }
 
 void PrestoExchangeSource::processDataResponse(
-    std::unique_ptr<http::HttpResponse> response) {
-  RECORD_HISTOGRAM_METRIC_VALUE(
-      kCounterExchangeRequestDuration, dataRequestRetryState_.durationMs());
-  RECORD_HISTOGRAM_METRIC_VALUE(
-      kCounterExchangeRequestNumTries, dataRequestRetryState_.numTries());
+    std::unique_ptr<http::HttpResponse> response, bool isGetDataSizeRequest) {
+  if (isGetDataSizeRequest) {
+    RECORD_HISTOGRAM_METRIC_VALUE(
+        kCounterExchangeGetDataSizeDuration,
+        dataRequestRetryState_.durationMs());
```

Comment:
Not sure how useful is the overall duration. Should we add a metric excluding the duration of a long pool? (waiting for pages).